pinot-plugins/pinot-stream-ingestion/pinot-pulsar/src/main/java/org/apache/pinot/plugin/stream/pulsar/PulsarConfig.java

```diff
@@ -46,6 +46,8 @@ public class PulsarConfig {
   private final String _authenticationToken;
   private final String _tlsTrustCertsFilePath;
   private final boolean _enableKeyValueStitch;
+  private final boolean _populateMetadata;
+  private final boolean _populateRecordTimeFromEventTime;
```

Comment:
nit: why is this configuration `_populateRecordTimeFromEventTime` needed ?

Reply:
@JeffBolle `StreamMessageMetadata#getRecordIngestionTimeMs()` is used in ingestion delay calculations. So, it would be incorrect to use user's eventTime as Pinot is trying to track the consumption from the stream and needs the time at which the record landed in the source. [See](https://github.com/apache/pinot/blob/68ab6bcd8da321960cbad4715ae80dd33ffd1d60/pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java#L1611C38-L1611C38)

Not super familiar with pulsar semantics - is `publishTime` set by the producer of the data or by the broker when it gets persisted in the stream? It should be the latter. 
In any case, you should avoid using event time as `recordIngestionTimeMs`

Reply:
I'll remove all of the config and code for setting the record time from event time.

From the documentation it appears that the `publishTime` field is populated by the producer running in user code when that producer sends the `Message` to the broker.  On the `Message` interface we also have the `brokerPublishTime`, which is documented as:
```
broker publish time from broker entry metadata, or empty if the feature is not enabled in the broker.
```
Since the broker field is optional, should we use it if present and fall back to the producer timestamp, or just always use the producer timestamp?

Reply:
> Since the broker field is optional, should we use it if present and fall back to the producer timestamp, or just always use the producer timestamp?

I think we can use the `brokerPublishTime` if present, else fallback to producer timestamp.